src/helpers/figures/charts/gauge_chart.ts

```diff
@@ -239,6 +264,13 @@ export class GaugeChart extends AbstractChart {
   getContextCreation(): ChartCreationContext {
     return {
       ...this,
+      title:
+        this.title.type === "reference"
+          ? this.title.reference
+            ? this.getters.getRangeString(this.title.reference, this.sheetId)
+            : ""
+          : this.title.value,
```

Comment:
you should always avoid using multi-level ternary statements. Since you ahve to repeat this ternary a bunch of times, I'd add a method (or helper) getTitle (do'nt forget to also adapt Odoo's PR)